Compare buildH JMelcr prog on the small Berger traj

patrickfuchs opened the pull request:
I added the comparison of the small Berger traj of 25 frames between buildH & JMelcr Prog. Conclusion : buildH rocks 😛 

Fix #57

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/OP_buildH.out:
```diff
@@ -0,0 +1,84 @@
+# OP_name            resname atom1 atom2  OP_mean OP_stddev OP_stem
```
We could remove this file since it's removed in the `launch.sh`

patrickfuchs replied:
Done!

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/OUT.progJMeclr.popc0-25ns_dt1000_H.out.line:
```diff
@@ -0,0 +1 @@
+conc  0.04829498499456276 0  0.07315715046570714 0    0.11729133781422464 0  0.14132863929036513 0
```
What is the purpose of this file?

patrickfuchs replied:
This is a file generated by the prog of Josef Melcr. Since we don't use it, I remove it.

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/README.md:
```diff
@@ -0,0 +1,10 @@
+In this dir, we demonstrate the validity of order parameter calculation with buildH, by comparing with the main program used in NMRlipids (written by J. Melcr). The detailed protocol is desribed in `launch.sh`.
```
**directory**

patrickfuchs replied:
Done

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/README.md:
```diff
@@ -0,0 +1,10 @@
+
+Practically, we launch buildH to reconstruct hydrogens on the Berger trajectory present in the parent directory.
+
+Then we launch JMelcr Prog to calculate the order parameters on this traj with Hs.
```
**program**

patrickfuchs replied:
Done

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/README.md:
```diff
@@ -0,0 +1,10 @@
+
+Then we launch JMelcr Prog to calculate the order parameters on this traj with Hs.
+
+Eventually, we compare the order parameters between the two programs. The results are plotted in `comp_JMelcrProg.pdf`. The mean difference averaged all order parameters is 0.0002. This very small difference is probably due to rounding errors, but it is by far completely negligible.
```
"averaged **over** all order paramaters"? I'm not sure here.

patrickfuchs replied:
Yeah I rephrased this. I meant it's averaged over all CHs.

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/calcOrderParametersPYTHON3.py:
```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python
```
We could remove this file since it's downloaded in the `launch.sh`

patrickfuchs replied:
Well, I wouldn't. If it changes in the future, at least we have the version I used. OK with you?

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/launch.sh:
```diff
@@ -0,0 +1,21 @@
+rm OP_buildH.out
+
+# Get prog from JMelcr on NMRlipids (last version in Python 3)
+wget https://raw.githubusercontent.com/NMRLipids/MATCH/master/scratch/scriptsBYmelcr/calcOrderParametersPYTHON3.py
```
Write in comment the date of the last modification of this script:
"Last edit 2017/02/02"

People will know against which version of this script, buildH has been compared.

patrickfuchs replied:
In fact this is not the real date I think. This date corresponds to the last change by Josef when it was in Python 2. Someone moved it to Python 3 but didn't write it there. It doesn't take to much space, again I would keep it here.

HubLot replied:
Oh okay. 
Maybe we can put the url of this file from the last commit of this file : 
https://raw.githubusercontent.com/NMRLipids/MATCH/3d1b8cf2068f41117f651cce69f95926513b5a19/scratch/scriptsBYmelcr/calcOrderParametersPYTHON3.py
Or keep this default url (pointing to master) and explain in the comment that the version downloaded correspond to this precise url (the one above, pointing to this specific commit)
In this way, we keep a trace of the 'correct' script.

I'm not really fan of keeping a "random" script in this manner but it's not a big deal either :)

patrickfuchs replied:
> I'm not really fan of keeping a "random" script in this manner but it's not a big deal either :)

Right, I agree with your point. But to me, it's no random script. I think it's essential  to keep a trace of the real script used to generate the data. If, for some reason, the repo hosting this script is destroyed (OK it's unlikely,  but it can happen sometime), then how can we prove the data were generated properly? Also, if for some reason, we want to regenerate the data, we're stuck.

HubLot replied:
Okay, let's keep it!

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/launch.sh:
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Launch buildH to reconstruct traj w Hs
+buildH ../start_128popc.pdb -l Berger_POPC -d ../order_parameter_definitions_MODEL_Berger_POPC.def -x ../popc0-25ns_dt1000.xtc -opx popc0-25ns_dt1000_H.xtc
```
"-opx **popc0-25ns_dt1000_H**"

patrickfuchs replied:
Right 😆 !

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/launch.sh:
```diff
@@ -0,0 +1,21 @@
+
+# Rename files
+mv popc0-25ns_dt1000_H.xtc.pdb start_128popc_H.pdb
+mv popc0-25ns_dt1000_H.xtc.xtc popc0-25ns_dt1000_H.xtc
```
Remove this line thanks to the above modification

HubLot commented on docs/Berger_POPC_test_case/comp_JMelcrProg/launch.sh:
```diff
@@ -0,0 +1,21 @@
+buildH ../start_128popc.pdb -l Berger_POPC -d ../order_parameter_definitions_MODEL_Berger_POPC.def -x ../popc0-25ns_dt1000.xtc -opx popc0-25ns_dt1000_H.xtc
+
+# Rename files
+mv popc0-25ns_dt1000_H.xtc.pdb start_128popc_H.pdb
```
"mv **popc0-25ns_dt1000_H.pdb** start_128popc_H.pdb"
(with the above modification)

patrickfuchs replied:
Done

HubLot reviewed (commented):
Great!  Now, the comparison is properly documented.

I think some files should be removed because they are either generated (popc0-25ns_dt1000_H.xtc, start_128popc_H.pdb, etc) or downloaded.

patrickfuchs commented:
Regarding `popc0-25ns_dt1000_H.xtc` and `start_128popc_H.pdb`, I would also leave them there. This is an example of complete output file generated by buildH. It's also another security check for us next to the tests. In principle, buildH should generate the same files in the future, so if it's stored here, we have a way to compare them. What do you think?

HubLot commented:
> Regarding `popc0-25ns_dt1000_H.xtc` and `start_128popc_H.pdb`, I would also leave them there. This is an example of complete output file generated by buildH. It's also another security check for us next to the tests. In principle, buildH should generate the same files in the future, so if it's stored here, we have a way to compare them. What do you think?

Okay but if I'm not mistaken, those 2 files are already present in the parent directory, aren't they?

patrickfuchs commented:
> Okay but if I'm not mistaken, those 2 files are already present in the parent directory, aren't they?

No, I removed them some time ago (can't recall in which commit). I think, their place is more suited in this dir `comp_JMelcrProg`.

HubLot commented:
> > Okay but if I'm not mistaken, those 2 files are already present in the parent directory, aren't they?
> 
> No, I removed them some time ago (can't recall in which commit). I think, their place is more suited in this dir `comp_JMelcrProg`.

I think they are still here : https://github.com/patrickfuchs/buildH/tree/master/docs/Berger_POPC_test_case
I'm okay keeping those in the `comp_JMelcrProg` directory as long as they are removed from the parent directory.

patrickfuchs commented:
> I think they are still here : https://github.com/patrickfuchs/buildH/tree/master/docs/Berger_POPC_test_case

Sorry, but I don't see `popc0-25ns_dt1000_H.xtc` or `start_128popc_H.pdb` there (on the master), do you?

> I'm okay keeping those in the `comp_JMelcrProg` directory as long as they are removed from the parent directory.

Sure!

HubLot commented:
oh i didn't see the "_H" at the end 😨 
My bad... 
Okay, we are good then :D

patrickfuchs commented:
Haha no worries 😆 !

patrickfuchs commented:
I realized that the initial script I downloaded was not giving the right OP_stdev (and OP_stem was not present). Thus I replaced by the most recent script I found [here](https://github.com/NMRLipids/MATCH/blob/master/scripts/calcOrderParameters.py). Looks like teh first script I used was random 🤣 ...kidding 😛 ! This new script was in Python 2, so I converted it to Python 3. Thus now we have a good reason to keep it in buildH repo 😉. Could you have a quick look @HubLot ? If you think it's OK, I'll merge this PR!

HubLot commented:
Good to merge!
